OSDOCS-1496 - Advanced networking tables

bobfuru opened the pull request:
[OSDOCS-1496](https://issues.redhat.com/browse/OSDOCS-1496) - Updates the advanced networking table (https://docs.openshift.com/container-platform/4.5/installing/installing_bare_metal/installing-bare-metal-network-customizations.html#installation-user-infra-machines-static-network_installing-bare-metal-network-customizations) to show options that can be added to the live installer boot prompt or the coreos-installer command line.

PREVIEW LINK: [Configure advanced networking](https://osdocs-1496--ocpdocs.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/installing-bare-metal-network-customizations.html#installation-user-infra-machines-static-network_installing-bare-metal-network-customizations)

Relates to https://github.com/openshift/openshift-docs/pull/25709

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -2,22 +2,37 @@
+This section illustrates the networking configuration options that allow you to modify the install process. The following tables describe the kernel arguments you can use with:
 
-.Configure advanced networking
+* The {op-system} live installer boot prompt by using routing and bonding for ISO installs.
+
+* The {op-system} live installer command by using `core.inst` for ISO or PXE installs.
+
+* The `coreos-installer` command for ISO installs.
```
Are these install refs for ISO/PXE correct?

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -2,22 +2,37 @@
+
+[IMPORTANT]
+====
+When using networking kernel arguments, you must specify `rd.neednet=1`.
```
This was a comment from Benjamin Gilbert. I don't know if this makes sense as a note, or if it should be included in each table. Also, is this required for each install type (boot prompt, `core.inst`, and `coreos-installer`?

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -2,22 +2,37 @@
+
+The following table describes how to use `ip=`, `nameserver=`, and `bond=` kernel arguments for ISO live installs.
+
+.Routing and bonding options for ISO
```
Does this wording make sense?

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,233 @@ enter the specific IP address you want and related information. For example:
+[discrete]
+=== `core.inst` command-line options for ISO or PXE install
+
+While you can pass most standard installation boot arguments to the live installer, there are several arguments that are specific to the {op-system} live installer.
```
_to the live installer_ - should we specify "to the RHEL live installer" here?

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,233 @@ enter the specific IP address you want and related information. For example:
+[source,adoc]
+|===
+|Option |Description
+
```
I left out a couple of options that referenced FCOS (`stream` and `stream-base-url`). OK?

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,233 @@ enter the specific IP address you want and related information. For example:
+|===
+
+[discrete]
+=== `coreos-installer` ISO Ignition examples
```
These were included towards the end of the G doc. I'm not sure if they should be included this way. And if so, would it make more sense to break out as a procedure module?

chrisnegus reviewed (commented):
LGTM. Thanks Bob!

sheriff-rh reviewed (approved):
Nice tables! LGTM! 🎉 🌮 🥇

bobfuru commented:
This will be merged to master and CP'ed to 4.6 so that it can be viewed and reviewed in the context of it's parent assembly,  https://github.com/openshift/openshift-docs/pull/25709.

Note that this PR 25656 has been reviewed by @chrisnegus but not reviewed by any other RHCOS SMEs.

bobfuru commented:
/cherrypick enterprise-4.6

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,199 @@ enter the specific IP address you want and related information. For example:
+
+a|`coreos.inst.image_url`
+
+a|Optional: Download and install the specified {op-system} image, overriding `coreos.inst.stream`.
```
Good point. I missed this mention of `coreos.inst.stream` and agree that it makes sense to remove it here.

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,199 @@ enter the specific IP address you want and related information. For example:
+
+* This argument should not be used in production environments and is intended for debugging purposes only.
+
+* You must use this option if you were already using the previous `coreos-installer`.
```
As long as there's nothing that an existing user must actually _do_ here when they upgrade from a previous version of OCP, then I agree it should be left out. Removed.

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,199 @@ enter the specific IP address you want and related information. For example:
+
+a|`coreos.inst.platform_id`
+
+a| Optional: The Ignition platform ID of the platform the {op-system} image is being installed on. Default is bare metal. This option determines whether or not to request an Ignition config from the cloud provider, such as AWS or WMware. For example: `coreos.inst.platform_id=vmware`.
```
Ack. Fixed the spelling errors and moved to the bottom of the table.

bgilbert replied:
VMware UPI should set this, so I'm +1 to leaving it in the doc.

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,199 @@ enter the specific IP address you want and related information. For example:
+
+a|`coreos.inst.insecure`
+
+a|Optional: Permits the OS image to be unsigned, and installs without doing a `gpgcheck`. Must be used with `coreos.inst.image_url`.
```
Resolved

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,199 @@ enter the specific IP address you want and related information. For example:
+
+a|`ignition.config.url`
+
+a|Identifies the location of an Ignition config that you build to pass selected values to be used by the live installer. This argument can be used for adding disk partitioning or other features you want to have in place before the actual installation runs.
```
Resolved. I also moved it to the bottom of the table so that the `coreos.inst` options are all listed first.

bobfuru commented:
@jlebon Thanks for the review and feedback! This PR had already been merged by the time you submitted your review, so I've addressed your comments in a new PR: https://github.com/openshift/openshift-docs/pull/26004. I'm going to get that new PR merged so that the changes are picked up by the overarching `coreos-installer` assembly in https://github.com/openshift/openshift-docs/pull/25709. 🎉

bgilbert commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,199 @@ enter the specific IP address you want and related information. For example:
+
+|*Command* |*Description*
+
+a|`$ coreos-installer iso ignition embed <options> <ISO_image> <Ignition_config>`
```
The `<Ignition_config>` arguments on these three commands are not correct.  Ignition configs, if specified, are passed via `-i/--ignition-file`.

bobfuru replied:
OK, I've updated this and the other two, for example: 
`$ coreos-installer iso ignition embed --ignition-file <options> <ISO_image>`

bgilbert commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,199 @@ enter the specific IP address you want and related information. For example:
+a|`coreos-installer iso ignition remove <options> <ISO_image> <Ignition_config>`
+a|Remove the embedded Ignition config from an ISO image.
+
+2+|_coreos-installer ISO Ignition options_
```
Not all of these are accepted by all subcommands.

bobfuru replied:
Added this as a note for this section.

bgilbert commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,199 @@ enter the specific IP address you want and related information. For example:
+
+|*Command* |*Description*
+
+a|`coreos-installer pxe ignition wrap <options> <initrd_name>`
```
There's no `initrd_name` positional parameter.

bobfuru replied:
So is the following revision correct?

```
a|`coreos-installer pxe ignition wrap <options> <image_name>
a|Wrap an Ignition config in an image.
```

bgilbert commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,199 @@ enter the specific IP address you want and related information. For example:
+
+2+|_coreos-installer PXE Ignition options_
+
+|*Option* |*Description*
```
These aren't accepted by all subcommands.

bobfuru replied:
Also added a note here:

> Note that not all of these options are accepted by all subcommands.

bgilbert reviewed (commented):
Apologies for the late review.

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -2,22 +2,37 @@
-.Configure advanced networking
+* The {op-system} live installer boot prompt by using routing and bonding.
+
+* The {op-system} live installer by using `core.inst` for ISO or PXE installs.
```
Good catch on the typos, thanks--fixed.

I've pared down the intro paragraph so that it is more generalized:

> This section illustrates the networking configuration and other advanced options that allow you to modify the RHCOS bare metal install process. The following tables describe the kernel arguments and command-line options you can use with the RHCOS live installer and the `coreos-installer` command.

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -2,22 +2,37 @@
 
-a|To configure an IP address, either use DHCP (`ip=dhcp`) or set an individual
-static IP address (`ip=<host_ip>`). Then identify the DNS server IP address (`nameserver=<dns_ip>`) on each node. This example sets: +
+a|To configure an IP address, either use DHCP (`ip=dhcp`) or set an individual static IP address (`ip=<host_ip>`). Then identify the DNS server IP address (`nameserver=<dns_ip>`) on each node. This example sets: +
```
I see the confusion. How about: 

> To configure an IP address, either use DHCP (`ip=dhcp`) or set an individual static IP address (`ip=<host_ip>`). _If setting a static IP, you must_ then identify the DNS server IP address (`nameserver=<dns_ip>`) on each node.

?

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,199 @@ enter the specific IP address you want and related information. For example:
+[discrete]
+== `core.inst` boot options for ISO or PXE install
+
+While you can pass most standard installation boot arguments to the live installer, there are several arguments that are specific to the {op-system} live installer.
```
OK. I've dropped "installation" so it reads, "While you can pass most standard boot arguments to the live installer..."

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,199 @@ enter the specific IP address you want and related information. For example:
+
+* For ISO, these options can be added by interrupting the {op-system} installer.
+
+* For PXE or iPXE, these options must be added to the `APPEND` line before starting the PXE kernel. You cannot interrupt a live PXE install.
```
I removed the second sentence ("You cannot interrupt...") to avoid this confusion. Does that make sense?

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,199 @@ enter the specific IP address you want and related information. For example:
+
+a|`coreos.inst.install_dev`
+
+a|Required. The block device on the system to install to. It is recommended to use the full path, such as `/dev/sda`.
```
It was suggested that I take `sda` only out but I've added back for context: 

> It is recommended to use the full path, such as `/dev/sda`, although `/sda` is allowed.

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,199 @@ enter the specific IP address you want and related information. For example:
+
+a|`coreos.inst.save_partindex`
+
+a|Optional: Comma-separated indexes of partitions to preserve during the install. Ranges `m`-`n` are permitted, and either `m` or `n` can be omitted. The specified partitions do not need to exist.
```
I wondered about that. I've changed to `m-n` -- not sure if you meant to have italics, too, but I left them out. Didn't look quite right.

bobfuru commented on modules/installation-user-infra-machines-static-network.adoc:
```diff
@@ -81,4 +97,199 @@ enter the specific IP address you want and related information. For example:
+[discrete]
+== `coreos-installer` options for ISO install
+
+You can run the `coreos-installer` command to identify various artifacts to include, to work with disk partitions, and to set up networking. In some cases, you can configure features on the live system and copy them to the installed system.
```
Sure. How about this?

> The `coreos-installer` command is another way for you to add boot options to the {op-system} live installer.
> 
> The kernel arguments in the previous table can provide a shortcut for invoking the `coreos-installer` program, but the `coreos-installer` command options are similar to those arguments, and you can run them directly from a shell prompt.
> 
> The following table shows the options and subcommands you can pass to the `coreos-installer` command during a live install.
